### PR TITLE
feat: add extra channel support to transform controller

### PR DESCRIPTION
**Pull request by utkuozdemir**
Add an option to specify an extra channel to the transform controller. Controller will wake up upon receiving data from the channel and run transform.

### CODE REVIEW

File: pkg/controller/generic/transform/controller.go [hunk 1]
```diff
@@ -198,6 +198,7 @@ func (ctrl *Controller[Input, Output]) Run(ctx context.Context, r controller.Run
 		case <-ctx.Done():
 			return nil
 		case <-r.EventCh():
+		case <-ctrl.options.extraCh:
```
Comment (frezbo) [inline pkg/controller/generic/transform/controller.go:201]: maybe call extraChInput?

File: pkg/controller/generic/transform/resource_test.go [hunk 2]
```diff
@@ -79,7 +79,8 @@ func (BE) ResourceDefinition() meta.ResourceDefinitionSpec {
 
 // BSpec provides B definition.
 type BSpec struct {
-	Out string
+	Out            string
+	TransformCount int
```
Comment (frezbo) [inline pkg/controller/generic/transform/resource_test.go:83]: don;t see code incrementing it?

File: pkg/controller/generic/transform/options.go [hunk 3]
```diff
@@ -70,3 +71,13 @@ func WithIgnoreTearingDownInputs() ControllerOption {
 		o.ignoreTearingDownInputs = true
 	}
 }
+
+// WithExtraChannel adds an extra channel to the controller.
+//
+// When this channel receives data, the controller will run the transform function.
+// This is useful to wake up the controller from a goroutine.
+func WithExtraChannel(extraCh <-chan struct{}) ControllerOption {
```
Comment (smira) [inline pkg/controller/generic/transform/options.go:79]:
```suggestion
func WithExtraEventChannel(extraEventCh <-chan struct{}) ControllerOption {
```

**Review by smira — APPROVED** [review]

File: pkg/controller/generic/transform/controller.go (hunk 1 above)
Comment (frezbo) [reply pkg/controller/generic/transform/controller.go]: just the variable name :sweat_smile:

**Review by frezbo — APPROVED** [review]

**Pull request merged**